src/sonic-dhcpservd/dhcpservd.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+
+import signal
+import json
+import syslog
+import psutil
+import time
+import ipaddress
+from dhcp_cfggen import DhcpServCfg
+from collections import deque
+from dhcp_server_utils import LEASE_FILE_PATH, DHCP_SERVER_IPV4_LEASE, DHCP_SERVER_IP_PORTS_FILE, \
+    INIT_CONFIG_FILE
+from dhcp_server_utils import DhcpDbConnector
+from functools import partial
+
+KEA_DHCP4_CONFIG = "/etc/kea/kea-dhcp4.conf"
+KEA_DHCP4_PROC_NAME = "kea-dhcp4"
+
+
+class DhcpServd(object):
+    def __init__(self, dhcp_cfg, db_connector):
+        self.dhcp_cfg = dhcp_cfg
+        self.db_connector = db_connector
+
+    def notify_kea_dhcp4_proc(self):
+        """
+        Send SIGHUP signal to kea-dhcp4 process
+        """
+        for proc in psutil.process_iter():
+            if KEA_DHCP4_PROC_NAME in proc.name():
+                proc.send_signal(signal.SIGHUP)
+                break
+
+    def generate_kea_dhcp4_config(self, from_db=True, config_file_path=""):
+        """
+        Generate kea-dhcp4 config file and dump it to config folder
+        Args:
+            from_db: boolean, if set to True, generate config from running config_db
+            config_file_path: str, if from_db is False, generate config from config_db file
+        """
+        kea_dhcp4_config = self.dhcp_cfg.generate_kea_dhcp4_config(from_db, config_file_path)
+        if kea_dhcp4_config is None:
+            syslog.syslog(syslog.LOG_ERR, "Cannot get kea-dhcp4 configure")
+            return
+        try:
+            with open(KEA_DHCP4_CONFIG, "w") as write_file:
+                json.dump(kea_dhcp4_config, write_file, indent=4, ensure_ascii=False)
+                # After refresh kea-config, we need to SIGHUP kea-dhcp4 process to read new config
+                self.notify_kea_dhcp4_proc()
+        except FileNotFoundError:
+            syslog.syslog(syslog.LOG_ERR, "Cannot write to: {}".format(INIT_CONFIG_FILE))
+
+    def update_lease_handler(self, signum, frame):
```

Comment:
`class LeaseManager(object)`
- list[] = LeaseHandler objects
- `start(self`) -> for obj in list: obj.register()
- `stop`

`class LeaseHandler(object)`
- `read(self)` abstract method
- `register(self)` abstract method
- `unregister`
- `update(self)` for lease in self.read() ... write state_db ...

`class KeaDhcp4LeaseHandler(LeaseHandler)`
- `read(self)` return list of leases
- `register(self)` -> `signal.signal(signal.SIGUSR1, self.update)`